internal/identity/api/routes.go

```diff
@@ -152,16 +152,23 @@ func (a *application) token(c *gin.Context) {
 	user, err := a.auth0.GetUserInfo(params.UserToken)
 
 	if err != nil {
-		log.WithError(err).Error("unauthorized token")
-		c.JSON(http.StatusUnauthorized, gin.H{"error": "unauthorized user"})
-		return
+		log.WithError(err).Error("failed to validate user token")
+		switch err := err.(type) {
```

Comment:
I like this idea, but I don't think we should have an Auth0UnexpectedError, let's leave that as default. What we should have an Unauthorized error and we return a StatusUnathorized back to the client, a RateLimitedError and we return a RateLimitedError back to the client, etc... probably a bad request should become an StatusInternalServerError as well

Reply:
Done